### ISSUE TITLE
[BUG] NullReferenceException on Error, when Response.Content.Headers is null

### ISSUE THREAD

**Issue by Echtelion** [reactiveui/refit#641]
When server respond an error (in my case 401), with an empty body content and without a ContentType, refit throw a NullReferenceException.
This is caused by :  
if (response.Content.Headers.ContentType.MediaType.Equals("application/problem+json"))

You should check for Headers, ContentType and MediaType beeing not null before trying to parse.

**Comment by Styxxy** [comment]
I can't reproduce this (with the master source code --> this is the latest published version). I added a test that should demonstrate this, though the test still succeeds. The "nullrefexception" is being swallowed.

See test: https://github.com/Styxxy/refit/blob/issue641-nullrefexception/Refit.Tests/ResponseTests.cs#L105

Can you give me a reproducible test for this issue, so I can look into it further?

**Comment by Echtelion** [comment]
Hi Styxxy,
The error was swallowed. I got the exception while debugging and I didn't see initially that it was a firstchanceexception, while the real exception was a http401. It was causing some mistake in understanding why I had a NullReferenceException in an expected test case, and wanted to be sure it was not from our request or the server response.
I checked the code, found this and report it to increase the quality code of refit.

Error management is not the best in class in refit, and errors thrown are not always easy to understand the root cause. I found this project very interesting, and it's a minor way to contribute.
